### PR TITLE
Enable encryption at rest

### PR DESCRIPTION
**Pull request by petr-balogh**
Fixes: #2695

Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

File: conf/ocsci/rest_encryption.yaml
Comment (ebenahar) [inline conf/ocsci/rest_encryption.yaml:6]: I think we should have it disabled by default

Comment (petr-balogh) [reply conf/ocsci/rest_encryption.yaml:6]: it will be disabled by default ;) Only when you pass this config file as described in MD file it will be enabled ;)

Comment (ebenahar) [reply conf/ocsci/rest_encryption.yaml:6]: ahh right!

Comment (petr-balogh) [reply conf/ocsci/rest_encryption.yaml:6]: Will be fixed here:  https://github.com/red-hat-storage/ocs-ci/issues/2699

**Review by fbalak — DISMISSED** [review]

File: conf/ocsci/rest_encryption.yaml
Comment (mbukatov) [inline conf/ocsci/rest_encryption.yaml:6]: Don't call it **rest encryption**, which could be little puzzling, but use `encryption_at_rest`.

**Review by mbukatov — CHANGES_REQUESTED** [review]
Requesting fix to use proper terminology.

**Comment by leseb** [comment]
How do we know if this code runs against RHCS 4.1z2 rc build? Thanks!

**Review by vavuthu — DISMISSED** [review]

**Review by mbukatov — APPROVED** [review]

**Comment by petr-balogh** [comment]
Opened BZ:  https://bugzilla.redhat.com/show_bug.cgi?id=1879072

**Comment by OdedViner** [comment]
You can add a deployment test that verify OSD is Encrypeted 

Test Procedure:
1.Go on the node where the OSD runs
2.Run "lsblk" command
3.Search the relevant line of "lsblk" command output (by name)
4.Search "crypt" string (from line on step 3)

[root@rook-ceph-osd-0-7cc8896cb7-mzr9x /]# lsblk 
NAME                              MAJ:MIN   RM   SIZE RO TYPE  MOUNTPOINT
xvda                              202:0      0   120G  0 disk  
|-xvda4                           202:4      0 119.5G  0 part  
| `-coreos-luks-root-nocrypt      253:0      0 119.5G  0 dm    /var/lib/ceph/crash
|-xvda2                           202:2      0   127M  0 part  
|-xvda3                           202:3      0     1M  0 part  
`-xvda1                           202:1      0   384M  0 part  
xvdcc                             202:20480  0    10G  0 disk  
`-set1-data-2-ks2qc-block-dmcrypt 253:1      0    10G  0 crypt 
loop0                               7:0      0    10G  0 loop  
xvdca                             202:19968  0     8G  0 disk

**Review by vasukulkarni — APPROVED** [review]

**Comment by petr-balogh** [comment]
> You can add a deployment test that verify OSD is Encrypeted
> 
> Test Procedure:
> 1.Go on the node where the OSD runs
> 2.Run "lsblk" command
> 3.Search the relevant line of "lsblk" command output (by name)
> 4.Search "crypt" string (from line on step 3)
> 
> [root@rook-ceph-osd-0-7cc8896cb7-mzr9x /]# lsblk
> NAME MAJ:MIN RM SIZE RO TYPE MOUNTPOINT
> xvda 202:0 0 120G 0 disk
> |-xvda4 202:4 0 119.5G 0 part
> | `-coreos-luks-root-nocrypt 253:0 0 119.5G 0 dm /var/lib/ceph/crash |-xvda2 202:2 0 127M 0 part |-xvda3 202:3 0 1M 0 part `-xvda1 202:1 0 384M 0 part
> xvdcc 202:20480 0 10G 0 disk
> `-set1-data-2-ks2qc-block-dmcrypt 253:1 0 10G 0 crypt
> loop0 7:0 0 10G 0 loop
> xvdca 202:19968 0 8G 0 disk

Hey @OdedViner , I will do in separate PR. As this will unblock the deployment ASAP. So I would merge this one and continue with this in other PR.

Opened issue: https://github.com/red-hat-storage/ocs-ci/issues/3035

**Comment by petr-balogh** [comment]
Currently blocked by: https://bugzilla.redhat.com/show_bug.cgi?id=1883927 

but the deployment without it enabled and this PR worked well and it actually it the BZ ;) so looks like everything should be fine here. So marking as verified. So we can merge and validation of encryption will be fixed in https://github.com/red-hat-storage/ocs-ci/issues/3035

**Comment by petr-balogh** [comment]
Verification job which is now running passed deployment and now tier1 is running. So now we can consider successful verification as well.

**Review by dahorak — APPROVED** [review]

**Review by mbukatov — APPROVED** [review]

**Pull request merged**
